### PR TITLE
Use non-blocking SecureRandom for embedded NETWORK_TOKEN_SECRETKEY

### LINKED ISSUE #9603: Use of SecureRandom.getInstanceStrong can cause database creation to block
## Expected behavior  
Database creation does not block on startup.

## Actual behavior  
When instantiating a database (e.g. an in-memory DB) the database creation blocks for a significant period (over 1 minute in our case).

## Steps to reproduce  
Create an embedded, in-memory, database without specifying a `NETWORK_TOKEN_SECRETKEY` configuration parameter.

## Root Cause Analysis
In 3.1, `OTokenHandlerImpl` used a `new RandomI()` to generate key bytes in the event that `NETWORK_TOKEN_SECRETKEY` was not defined in the configuration.
This was non-blocking (good) but not a secure PRNG (bad).

In 3.2 this has been changed (in `OTokenSignImpl`) to use `SecureRandom.getInstanceStrong()`, which (on Linux and Mac at least) defaults to the `NativePRNGBlocking` implementation, which uses the (blocking) `/dev/random` device.

A static `new SecureRandom()`, which will choose a secure non-blocking PRNG, would be the right thing to use in this scenario.

This issue has come up in a bunch of projects:
- https://tersesystems.com/blog/2015/12/17/the-right-way-to-use-securerandom/ explains what is going on in the SecureRandom
- https://github.com/akka/akka/issues/20908 is an example of Akka making this change

### PR DESCRIPTION
**Pull request by timw**
Avoid embedded database creation blocking due to unnecessary use of strong SecureRandom.getInstanceStrong.

Fixes #9603

### CODE REVIEW

File: core/src/main/java/com/orientechnologies/orient/core/security/OTokenSignImpl.java [hunk 1]
```diff
@@ -135,14 +134,7 @@ public String getDefaultKey() {
     if (configKey != null && configKey.length() > 0) key = Base64.getUrlDecoder().decode(configKey);
 
     if (key == null) {
-      try {
-        key =
-            OSecurityManager.digestSHA256(
-                String.valueOf(SecureRandom.getInstanceStrong().nextLong()));
-      } catch (NoSuchAlgorithmException e) {
```
Comment (mmacfadden) [inline core/src/main/java/com/orientechnologies/orient/core/security/OTokenSignImpl.java:142]: Is there a reason we have taken away the try catch?  It seems like NoSuchAlgorithmException won't actually be thrown here I guess.  However looking at the code for digestSHA256, it looks like a OConfigurationException could be thrown.  Should we catch that, still, and re-throw as before?

**Review by mmacfadden — COMMENTED** [review]
I read the various article and references on this topic, and this looks like a reasonable and correct change.  I did notice that some exception handling code was change instead of just changing the random generation. While it doesn't look like that should cause a problem and was probably just clean up. Could it be safer to just leave that and just make the single line change?

File: core/src/main/java/com/orientechnologies/orient/core/security/OTokenSignImpl.java (hunk 1 above)
Comment (timw) [reply core/src/main/java/com/orientechnologies/orient/core/security/OTokenSignImpl.java:142]: Only NoSuchAlgorithmException was caught here before, and is no longer possible, so removed handler for it.

**Comment by luigidellaquila** [comment]
Hi @timw 

That seems good, thank you very much! Merging

Luigi

**Pull request merged**